File: android-app/app/src/main/java/com/example/androidapp/authentication/Validation.java
package com.example.androidapp.authentication;

import android.content.Context;
import android.text.TextUtils;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.Toast;

public class Validation {

    public static boolean validateUsername(EditText username) {
        int MIN_USERNAME_LENGTH = 4;
        if (username.getText().toString().length() < MIN_USERNAME_LENGTH) {
            username.setError("Username needs to have at least 4 characters");
            return false;
        }
        return true;
    }

    public static boolean validatePassword(EditText password) {
        int MIN_PASSWORD_LENGTH = 8;
        if (password.getText().toString().length() < MIN_PASSWORD_LENGTH) {
            password.setError("Password needs have at least 8 characters");
            return false;
        }
        return true;
    }

    public static boolean confirmPasswordsMatch(EditText password, EditText confirmPassword) {
        if (!confirmPassword.getText().toString().equals(password.getText().toString())) {
            confirmPassword.setError("Passwords do not match");
            return false;
        }
        return true;
    }

    public static boolean validateDisplayName(EditText displayName) {
        if (TextUtils.isEmpty(displayName.getText())) {
            displayName.setError("Display Name can't be empty");
            return false;
        }
        return true;
    }

    public static boolean validateProfilePicture(ImageView profilePicture, Context context) {
        if (profilePicture.getDrawable() == null) {
            CharSequence text = "Image is required";
            int duration = Toast.LENGTH_SHORT;

            Toast toast = Toast.makeText(context, text, duration);
            toast.show();
            return false;
        }
        return true;
    }
}

File: react-app/src/Authentication/Validation.ts
const MIN_USERNAME_LENGTH = 4;
const MIN_PASSWORD_LENGTH = 8;

/** Returns an error message, or null when the value is valid. */
export function validateUsername(username: string): string | null {
	if (username.length < MIN_USERNAME_LENGTH) {
		return `Username needs to have at least ${MIN_USERNAME_LENGTH} characters`;
	}
	return null;
}

export function validatePassword(password: string): string | null {
	if (password.length < MIN_PASSWORD_LENGTH) {
		return `Password needs have at least ${MIN_PASSWORD_LENGTH} characters`;
	}
	return null;
}

export function validateConfirmPassword(confirmPassword: string, password: string): string | null {
	if (confirmPassword.length < MIN_PASSWORD_LENGTH) {
		return `Password needs have at least ${MIN_PASSWORD_LENGTH} characters`;
	}
	if (confirmPassword !== password) {
		return "Passwords do not match";
	}
	return null;
}

export function validateDisplayName(displayName: string): string | null {
	if (displayName.length === 0) {
		return "Display Name can't be empty";
	}
	return null;
}
